pkg/operator/options/options.go

```diff
@@ -59,23 +86,85 @@ func New() *Options {
 	f.BoolVar(&opts.EnableLeaderElection, "leader-elect", env.WithDefaultBool("LEADER_ELECT", true), "Start leader election client and gain leadership before executing the main loop. Enable this when running replicated components for high availability.")
 	f.Int64Var(&opts.MemoryLimit, "memory-limit", env.WithDefaultInt64("MEMORY_LIMIT", -1), "Memory limit on the container running the controller. The GC soft memory limit is set to 90% of this value.")
 
+	// Vars that must be merged with settings
+	f.DurationVar(&opts.BatchMaxDuration, "batch-max-duration", env.WithDefaultDuration("BATCH_MAX_DURATION", 10*time.Second), "The maximum length of a batch window. The longer this is, the more pods we can consider for provisioning at one time which usually results in fewer but larger nodes.")
+	f.DurationVar(&opts.BatchIdleDuration, "batch-idle-duration", env.WithDefaultDuration("BATCH_IDLE_DURATION", time.Second), "The maximum amount of time with no new pending pods that if exceeded ends the current batching window. If pods arrive faster than this time, the batching window will be extended up to the maxDuration. If they arrive slower, the pods will be batched separately.")
+	f.StringVar(&opts.featureGates, "feature-gates", env.WithDefaultString("FEATURE_GATES", ""), "Optional features can be enabled / disabled using feature gates. Current options are: Drift")
+
 	if opts.MemoryLimit > 0 {
 		newLimit := int64(float64(opts.MemoryLimit) * 0.9)
 		debug.SetMemoryLimit(newLimit)
 	}
+
 	return opts
 }
 
-// MustParse reads the user passed flags, environment variables, and default values.
-// Options are valided and panics if an error is returned
 func (o *Options) MustParse() *Options {
-	err := o.Parse(os.Args[1:])
-
-	if errors.Is(err, flag.ErrHelp) {
+	if err := o.Parse(os.Args[1:]); errors.Is(err, flag.ErrHelp) {
 		os.Exit(0)
-	}
-	if err != nil {
+	} else if err != nil {
 		panic(err)
 	}
+
+	o.mustParseFeatureGates()
+
+	// Check if BatchMaxDuration and BatchIdleDuration have been set. If they have not been set, they will be overwritten by
+	// values in the configmap, if present.
+	o.Visit(func(f *flag.Flag) {
+		switch f.Name {
+		case "batch-max-duration":
+			o.BatchMaxDurationSet = true
+		case "batch-idle-duration":
+			o.BatchIdleDurationSet = true
+		}
+	})
+	if _, ok := os.LookupEnv("BATCH_MAX_DURATION"); ok {
+		o.BatchMaxDurationSet = true
+	}
+	if _, ok := os.LookupEnv("BATCH_IDLE_DURATION"); ok {
+		o.BatchIdleDurationSet = true
+	}
+
 	return o
 }
+
+// MergeSettings applies settings specified in the v1alpha5 configmap to options. If the value was already specified by
+// a CLI argument or environment variable, that value will be used.
+func (o *Options) MergeSettings(s *settings.Settings) {
+	mergeField(&o.BatchMaxDuration, s.BatchMaxDuration, o.BatchMaxDurationSet)
```

Comment:
Does this still retain default values? I think the answer is still no if the CLI argument isn't set and the settings value isn't set. You still get a scenario where you have a default value that's stored from the CLI options and you override it with the setting every time, whether or not the setting value is set

Reply:
Since settings also has default values applied to it, if neither are set it will apply the default from settings. The downside of this approach is that the defaults are defined in two places but the upside is there's no additional logic that needs to be torn out with settings once support is dropped.